File: js-archive/reformat-date.js
const reformatDate = function (date) {
  date = date.split(' ').reverse()
  const mem = {
    "Jan": 1,
    "Feb": 2,
    "Mar": 3,
    "Apr": 4,
    "May": 5,
    "Jun": 6,
    "Jul": 7,
    "Aug": 8,
    "Sep": 9,
    "Oct": 10,
    "Nov": 11,
    "Dec": 12
  }
  date[1] = mem[date[1]]
  date[2] = date[2].replace(/[a-zA-Zа-яА-Я]/g, '')
  date[1] = addNull(date[1])
  date[2] = addNull(date[2])
  return date.join('-')
}

function addNull(str) {
  if (typeof str !== 'string') {
    str += ''
  }
  if (str.length < 2) {
    str = '0' + str
  }

  return str
}

console.log(reformatDate("6th Jun 1933"))

File: go-archive/reformat-date.go
func reformatDate(date string) string {
    s:= strings.Split(date, " ")
    d:=s[0]; m:=s[1]; y:=s[2];

    mtn:=map[string]string{
        "Jan":"01",
        "Feb":"02",
        "Mar":"03",
        "Apr":"04",
        "May":"05",
        "Jun":"06",
        "Jul":"07",
        "Aug":"08",
        "Sep":"09",
        "Oct":"10",
        "Nov":"11",
        "Dec":"12",
    }
    d = day(d)
    y+="-" + mtn[m] + "-" + d

    return y
}

func day (d string) string {
    ans:=""

    for i:=0; i < len(d); i++ {
        if v, err := strconv.Atoi(string(d[i])); err == nil {
            ans+=strconv.Itoa(v)
        }
    }

    if (len(ans) < 2) {ans = "0" + ans}
    return ans
}